scripts/Interface/ui/drill/model/DrillExportModel.py

```diff
@@ -0,0 +1,206 @@
+            return True
+        criteria = RundexSettings.EXPORT_ALGO_CRITERIA[algo]
+        if not criteria:
+            return True
+
+        processingAlgo = mtd[ws].getHistory().lastAlgorithm()
+        try:
+            params = re.findall("%[a-zA-Z]*%", criteria)
+            for param in params:
+                value = processingAlgo.getPropertyValue(param[1:-1])
+                criteria = criteria.replace(param, '"' + value + '"')
+            return bool(eval(criteria))
+        except:
+            return False
+
+    def _onTaskSuccess(self, name):
+        """
+        Triggered when the export finished with success.
+
+        Args:
+            name (str): the task name
+        """
+        name = name.split(':')
+        wsName = name[0]
+        filename = name[1]
+
+        if wsName not in self._successExports:
+            self._successExports[wsName] = set()
+        self._successExports[wsName].add(filename)
+
+        if wsName in self._exports:
+            self._exports[wsName].discard(filename)
+            if not self._exports[wsName]:
+                del self._exports[wsName]
+                self._logSuccessExport(wsName)
+
+    def _onTaskError(self, name, msg):
+        """
+        Triggered when the export failed.
+
+        Args:
+            name (str): the task name
+            msg (str): error msg
+        """
+        name = name.split(':')
+        wsName = name[0]
+        filename = name[1]
+
+        logger.error("Error while exporting workspace {}.".format(wsName))
+        logger.error(msg)
+
+        if wsName in self._exports:
+            self._exports[wsName].discard(filename)
+            if not self._exports[wsName]:
+                del self._exports[wsName]
+                self._logSuccessExport(wsName)
+
+    def _logSuccessExport(self, wsName):
+        """
+        Log all the successful exports.
+
+        Args:
+            wsName (str): name of the concerned workspace
+        """
+        if wsName not in self._successExports:
+            return
+        filenames = ", ".join(self._successExports[wsName])
+        logger.notice("Successful export of workspace {} to {}"
+                      .format(wsName, filenames))
+        del self._successExports[wsName]
+
+    def run(self, sample):
+        """
+        Run the export algorithms on a workspace. If the provided workspace is
+        a groupworkspace, the export algorithms will be run on each member of
+        the group.
+
+        Args:
+            workspaceName (str): name of the workspace
+        """
+        exportPath = config.getString("defaultsave.directory")
+        workspaceName = sample.getOutputName()
+
+        tasks = list()
+        for wsName in mtd.getObjectNames():
+            if ((workspaceName in wsName)
+                    and (not isinstance(mtd[wsName], WorkspaceGroup))):
+                for a,s in self._exportAlgorithms.items():
+                    if s and self._validCriteria(wsName, a):
```

Comment:
I think it would be nice to log a debug message saying "export skipped" or similar if the algorithm doesn't meet the criteria

Reply:
I added a log in the notice channel for this.